include/hermite_multidimensional.hpp

```diff
@@ -46,6 +46,44 @@ ullint vec2index(std::vector<int> &pos, int resolution) {
 
 }
 
+/**
+ * Updates the iterators needed for the calculation of the Hermite multidimensional functions
+ *
+ * @param nextPos a vector of integers
+ * @param jumpFrom a vector of integers
+ * @param jump integer specifying whether to jump to the next index
+ * @param resolution integer specifying the cuotff
+ * @dim dimension of the R matrix
+ *
+ * @k index necessary for knowing which elements are needed from the input vector y and matrix R
+ */
+int update_iterator(std::vector<int> &nextPos, std::vector<int> &jumpFrom, int &jump, const int &resolution, const int &dim) {
+    if (jump > 0) {
+        jumpFrom[jump] += 1;
+        jump = 0;
+    }
+    for (int ii = 0; ii < dim; ii++) {
+        std::vector<int> forwardStep(dim, 0);
+        forwardStep[ii] = 1;
+
+        if ( forwardStep[ii] + nextPos[ii] > resolution) {
+            nextPos[ii] = 1;
+            jumpFrom[ii] = 1;
+            jump = ii+1;
+        }
+        else {
+            jumpFrom[ii] = nextPos[ii];
+            nextPos[ii] = nextPos[ii] + 1;
+            break;
```

Comment:
See other pull request review for an optimization here (sorry accidentally put that comment in the wrong pull request).

Reply:
nevermind